servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -194,6 +197,15 @@ public void channelWritable() {
     @Override
     public void channelOutboundClosed() {
         assert eventLoop.inEventLoop();
+        final Subscription sub = this.subscription;
+        if (sub != null) {
+            // Some protocols may know the source should terminate before the user is done writing all the content. For
+            // example if HTTP headers includes content-length: 0, the protocol knows it is done after writing the
+            // metadata, but the user may have also written empty buffer and/or empty trailers that still need to be
+            // consumed. If the FlushStrategy waits until the stream terminates we must request the remainder of content
+            // we may deadlock if we don't request enough onNext signals to see the terminal signal.
+            sub.request(Long.MAX_VALUE);
```

Comment:
@idelpivnitskiy - note this, not ideal but happy to discuss alternatives if you have ideas...